### DISCUSSION TITLE
Get data from action in another route

### DISCUSSION THREAD

**Discussion by nathanlindsay** [remix-run/remix#869] [Q&A] [answered]
If I have a `<Form action="/some-other-route">`, how do I get the return value from that action?

`useActionData()` only seems to work for actions in the same route...

**Comment by QuentinWidlocher** [comment]
I think you can use a fetcher (useFetcher()) to make a form with a GET method.
That allows getting data on another route without reloading the page

**Comment by sergiodxa** [comment]
You need to move the action to the same route where you can useLoaderData or use session.flash so save the action data and redirect back to the url of the form, then read the data from the session inside the loader of that route.

**Comment by ryanflorence** [comment, marked as answer]
Like @sergiodxa said, you need to

- move the action code to the route (this is what I would do, since you want the data there, move the action there)
- redirect back to the page with the form and store state in a cookie, read it in the loader (pain in the neck but works with SSR, no JS, etc.)
- `useFetcher` - doesn't cause navigation, doesn't server render, doesn't work without JS but probably matches your current mental model for this feature
